webapp/src/shared/components/Game/state.tsx

```diff
@@ -0,0 +1,142 @@
+import { getBallPos, getPaddlePos } from './physics';
+import { GameBall, GamePaddle, Coord } from './models';
+import {
+  CANVAS_WIDTH,
+  CANVAS_HEIGHT,
+  PADDLE_WIDTH,
+  PADDLE_HEIGHT,
+  PADDLE_SLIDE_SPEED,
+  BALL_RADIUS,
+  BALL_SPEED,
+} from './constants';
+
+export type GameState = {
+  ball: GameBall;
+  paddle: GamePaddle;
+  score: number;
+};
+
+type Act<Type extends string, Payload extends {}> = {
+  type: Type;
+  payload: Payload;
+};
+
+export type Action =
+  | Act<'move', { deltaTime: number }>
+  | Act<
+      'lose' | 'win' | 'paddleMoveRight' | 'paddleMoveLeft' | 'paddleStop',
+      {}
+    >
+  | Act<'paddleDrag', { dragCurrPos: number; dragPrevPos: number }>;
+
+const getPaddleDragX = (
+  paddle: GamePaddle,
+  dragPrevPos: number,
+  dragCurrPos: number,
+): number => {
+  const deltaX = paddle.x - (dragPrevPos - dragCurrPos);
+
+  if (deltaX < 0) {
+    return 0;
+  } else if (deltaX > CANVAS_WIDTH - PADDLE_WIDTH) {
+    return CANVAS_WIDTH - PADDLE_WIDTH;
+  }
+  return deltaX;
+};
+
+const calcInitialBallSpeed = (): Coord => {
+  const coef = Math.ceil(3 * Math.random());
+  const angle = (Math.PI / 2) * coef + Math.PI / 4;
+
+  return {
+    x: BALL_SPEED * Math.cos(angle),
+    y: BALL_SPEED * Math.sin(angle),
+  };
+};
+
+export const initialBallState = (): GameBall => {
```

Comment:
Where should this eventually go? Here or physics? Now its more duplicated...

Reply:
This should go in the state, it was a leftover, deleted!